src/components/Transactions/TransactionGeneric.tsx

```diff
@@ -0,0 +1,174 @@
+import { Box, Collapse, Flex, Link, SimpleGrid } from '@chakra-ui/react'
+import dayjs from 'dayjs'
+import { useState } from 'react'
+import { FaStickyNote } from 'react-icons/fa'
+import { Amount } from 'components/Amount/Amount'
+import { IconCircle } from 'components/IconCircle'
+import { MiddleEllipsis } from 'components/MiddleEllipsis/MiddleEllipsis'
+import { Row } from 'components/Row/Row'
+import { RawText, Text } from 'components/Text'
+import { TransactionStatus } from 'components/Transactions/TransactionStatus'
+import { TxDetails } from 'hooks/useTxDetails/useTxDetails'
+import { fromBaseUnit } from 'lib/math'
+
+export const TransactionGeneric = ({ txDetails }: { txDetails: TxDetails }) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const toggleOpen = () => setIsOpen(!isOpen)
+
+  const toAddress = txDetails.ensTo || txDetails.to
+  const fromAddress = txDetails.ensFrom || txDetails.from
+
+  return (
+    <>
+      <Flex
+        alignItems='center'
+        flex={1}
+        justifyContent='space-between'
+        textAlign='left'
+        as='button'
+        w='full'
+        py={4}
+        onClick={toggleOpen}
```

Comment:
Hey ✋ 

I have a doubt, because of this, when you click on the tx id, will it close the `Collapse`?

Reply:
hey @NeOMakinG 👋 

It will not close when clicking on the transaction ID hyperlink, or when clicking anywhere in the expanded component (this matches the behaviour of our existing transaction components).

Does that answer the question?

Reply:
Thanks for the answer :)